docs/3.0/JSX.md

```diff
@@ -0,0 +1,114 @@
+---
+---
+# JSX
+
+JSX is a proprietary extension to the JavaScript/TypeScript syntax that allows mixing code with XML-like declarations. Tabris 3 supports JSX out of the box in `.jsx` and `.tsx` files with any TypeScript compiler based projects. The only requirement is that in `tsconfig.json` the
+`jsx` property is set to `"react"` and `jsxFactory` to `"JSX.createElement"`.
+
+## Usage
+
+In Tabris, JSX is used to create UI elements in a declarative manner. Every constructor for a Widget, WidgetCollection or Popup based class can be used as an JSX element. (As of 3.0-beta1 only the ActionSheet popup supports JSX. This will change in the final release.)
+
+Action | JavaScript/TypeScript | JSX
+---|---|---
+Create an instance|`new TextView()`|`<TextView />`
+Set a string property*|`new TextView({text: 'foo'})`|`<TextView text='foo' />`
+Set non-string property*|`new TextView({elevation: 3})`|`<TextView elevation={3} />`
+Set property to true|`new TextView({markupEnabled: true})`|`<TextView markupEnabled />`
+Register listener|`new TextView().onResize(cb)`|`<TextView onResize={cb} />`
+Set properties object|`new TextView(props)`|`<TextView {...props} />`
+Mix in properties object|`new TextView({text: 'foo', ...props})`|`<TextView text='foo' {...props} />`
+Add new children|`new Composite().append(new TextView())`|`<Composite><TextView /></Composite>`
+Add existing children|`new Composite().append(children)`|`<Composite>{children}</Composite>`
+
+\* Some properties can also be set by putting the value between the opening and closing tag, like children. For `<TextView>foo</TextView>` sets the `text` property to `'foo'`. Properties that support this are marked as such in the API reference.
+
+> :warning: In TypeScript JSX expressions themselves are type-safe, but their return type is `any`! So be extra careful when you assign them to a variable to give it the proper type.
+
+To add multiple children to an existing parent you group them using `WidgetCollection`:
+
+```jsx
+// JavaScript/TypeScript:
+contentView.append(
+  new Button(),
+  new TextView()
+);
+
+// JSX:
+contentView.append(
+  <WidgetCollection>
+    <Button />
+    <TextView />
+  </WidgetCollection>
+);
+```
+
+This is not necessary inside JSX elements:
+
+```xml
+<Composite>
+  <Button />
+  <TextView />
+</Composite>
+```
+
+## Custom JSX Elements
```

Comment:
I plan to extract everything below in a separate article so we have one "custom component" article for each technology (JS, JSX, TS+Decorators)

Reply:
I opened an in issue for further discussion.